examples/s/d/nn/_project/aggregate_surfaces.py

```diff
@@ -0,0 +1,195 @@
+        surfacepath = casepath / f"realization-{r}" / iter_name / relative_path
+        surface = xtgeo.surface_from_file(surfacepath)
+
+        metadatapath = _metadata_filename(surfacepath)
+        metadata = _parse_yaml(metadatapath)
+
+        # this example is minimalistic and super non-robust. In reality, there will
+        # be realizations missing etc which needs to be handled.
+
+        collected_data.append((surface, metadata))
+
+    source_surfaces = xtgeo.Surfaces([surface for surface, _ in collected_data])
+    source_metadata = [metadata for _, metadata in collected_data]
+
+    return source_surfaces, source_metadata
+
+
+def main():
+    """Aggregate one surface across X realizations from the example case."""
+
+    # get the input realisations
+    casepath = Path("examples/s/d/nn/xcase/").resolve()
+
+    # Assuming that we know the surface name here.
+    # Usually, this would be derived from a wildcard search on realization-0 or similar
+    # The point is that each realization contains a representation of the same surface
+    # and it has the same filename across all realizations.
+
+    iter_name = "iter-0"
+    relative_path = (
+        "share/results/maps/topvolantis--ds_extract_geogrid.gri"  # exists in all reals
+    )
+
+    reals = _get_realizations(casepath)
+
+    # gather source surfaces and their associated metadata
+    source_surfaces, source_metadata = _get_source_surfaces_from_disk(
+        casepath=casepath, iter_name=iter_name, reals=reals, relative_path=relative_path
+    )
+
+    # Initialize a dictionary in which we will gather all the results
+    aggregations = {
+        "mean": {
+            "operation": "mean",
+            "surface": source_surfaces.apply(np.nanmean, axis=0),
+        },
+        "min": {"operation": "min", "surface": source_surfaces.apply(np.min, axis=0)},
+        "max": {"operation": "max", "surface": source_surfaces.apply(np.max, axis=0)},
+        "std": {
+            "operation": "std",
+            "surface": source_surfaces.apply(np.std, axis=0),
+        }
+        # percentiles are time-consuming to do directly, so need separate functions
+        # for this which is currently not included in the example. #TODO
+    }
+
+    # now populate with metadata
+    for agg in aggregations:
+        aggregation = aggregations[agg]
+        operation = aggregation["operation"]
+        surface = aggregation["surface"]
+        aggregation_id = "0000-0000"  # an identifier for these aggregations
+
+        # TODO: Check if surfaces are re-calculated here...
+
+        exp = fmu.dataio.ExportAggregatedData(
+            source_metadata=source_metadata,  # list of metadata for the surfaces used
+            aggregation_id=aggregation_id,
+            verbosity="DEBUG",
+        )
+        # We generate metadata and store in the dictionary for later usage
+        metadata = exp.generate_metadata(obj=surface, operation=operation)
+        aggregation["metadata"] = metadata
+
```

Comment:
Do we need kwarg `obj=surface`? or could we live with arg here as in `ExportData.export()`

`metadata = exp.generate_metadata(surface, operation=operation)`

Reply:
No, not needed. See #192